soc/arm/nordic_nrf/nrf53/soc.c

```diff
@@ -89,5 +89,31 @@ void z_platform_init(void)
 	SystemInit();
 }
 
+#if defined(CONFIG_SOC_NRF5340_CPUAPP) && \
+	!defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
+bool nrf53_has_erratum19(void)
+{
+	if (NRF_FICR->INFO.PART == 0x5340) {
+		if (NRF_FICR->INFO.VARIANT == 0x41414142) {
+			return true;
+		}
+	}
+	return false;
+}
+
+#ifndef NRF5340_CPUAPP_ERRATUM19
+static int check_erratum19(struct device *arg)
+{
+	ARG_UNUSED(arg);
+	__ASSERT(!nrf53_has_erratum19(),
```

Comment:
This does not seem safe, if one disables ASSERTS. We cannot really add a k_panic at this very early stage of booting, either. We could move this runtime checking at some point later in the code, when the zephyr kernel is initialized.
Any ideas?

Reply:
Oh, i see, now , there's a POST_KERNEL SYS_INIT, for the erratum check, sorry I missed that

Reply:
Added error log and k_panic()